src/core/api/project.go

```diff
@@ -712,3 +707,34 @@ func getProjectMemberSummary(projectID int64, summary *models.ProjectSummary) {
 
 	wg.Wait()
 }
+
+// Returns the highest role in the role list.
+// This func should be removed once we deprecate the "current_user_role_id" in project API
+// A user can have multiple roles and they may not have a strict ranking relationship
+func highestRole(roles []int) int {
+	if roles == nil {
+		return 0
+	}
+	rolePower := map[int]int{
+		common.RoleProjectAdmin: 50,
+		common.RoleMaster:       40,
+		common.RoleDeveloper:    30,
+		common.RoleGuest:        20,
+		common.RoleLimitedGuest: 10,
+	}
+	var ind int
+	for _, r := range roles {
+		if _, ok := rolePower[r]; ok {
+			roles[ind] = r
+			ind++
+		}
+	}
+	roles = roles[:ind]
+	if len(roles) == 0 {
+		return 0
+	}
+	sort.Slice(roles, func(x, y int) bool {
+		return rolePower[roles[x]] > rolePower[roles[y]]
+	})
+	return roles[0]
```

Comment:
Maybe compare role one by one will be better than sort.

```go
var ( 
        highest
        highestPower
) int

for _, role := range roles {
        power := rolePower[role]
        if power > highestPower {
               highest = role
               highestPower  = power
        }
}

return highest
```

And line 715-717 not needed when compare role one by one.

Reply:
leaving 715-717 may just make it a little bit more robust?